### PR TITLE
Test: Support coprocessor test

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #4609 

Problem Summary:

### What is changed and how it works?
Add test API to run Coprocessor.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/FlashService.cpp [hunk 1]
```diff
@@ -430,8 +433,8 @@ std::tuple<ContextPtr, grpc::Status> FlashService::createDBContext(const grpc::S
         std::string client_ip = peer.substr(pos + 1);
         Poco::Net::SocketAddress client_address(client_ip);
 
-        // For MPP test, we don't care about security config.
-        if (!context->isMPPTest())
+        // For MPP or Cop test, we don't care about security config.
+        if (!(context->isMPPTest() || context->isCopTest()))
```
Comment (SeaRise) [inline dbms/src/Flash/FlashService.cpp:437]:
```suggestion
        if (unlikely(!(context->isMPPTest() || context->isCopTest())))
```

File: dbms/src/TestUtils/MPPTaskTestUtils.cpp [hunk 2]
```diff
@@ -84,6 +86,46 @@ ColumnsWithTypeAndName MPPTaskTestUtils::exeucteMPPTasks(QueryTasks & tasks, con
...
+    TiFlashTestEnv::getGlobalContext(test_meta.context_idx).setCopTest();
+
+    MockComputeServerManager::instance().setMockStorage(context.mockStorage());
+
+    auto addr = MockComputeServerManager::instance().getServerConfigMap()[test_meta.context_idx - 1].addr;
+    MockComputeClient client(
+        grpc::CreateChannel(addr, grpc::InsecureChannelCredentials()));
+    auto resp = client.runCoprocessor(req);
+    auto resp_ptr = std::make_shared<tipb::SelectResponse>();
+    if (!resp_ptr->ParseFromString(resp.data()))
```
Comment (SeaRise) [inline dbms/src/TestUtils/MPPTaskTestUtils.cpp:119]:
```suggestion
    if (unlikely(!resp_ptr->ParseFromString(resp.data())))
```

File: dbms/src/Interpreters/Context.h [hunk 3]
```diff
@@ -161,6 +161,7 @@ class Context
     {
         non_test,
         mpp_test,
+        cop_test,
```
Comment (SeaRise) [inline dbms/src/Interpreters/Context.h:164]: It looks like `mpp_test` and `cop_test` can be combined into one `mock_server_test`.

File: dbms/src/Flash/FlashService.cpp (hunk 1 above)
Comment (ywqzzy) [reply dbms/src/Flash/FlashService.cpp:437]: likely

**Comment by ywqzzy** [comment]
Note that if we want to make the coprocessor query run indefinitely, we can change MockSourceStream.h:65 to construct a infinite TableScanStream.

File: dbms/src/Flash/FlashService.cpp [hunk 4]
```diff
@@ -105,7 +105,8 @@ grpc::Status FlashService::Coprocessor(
     CPUAffinityManager::getInstance().bindSelfGrpcThread();
     LOG_FMT_DEBUG(log, "Handling coprocessor request: {}", request->DebugString());
 
-    if (!security_config->checkGrpcContext(grpc_context))
+    // For coprocessor test, we don't care about security config.
+    if (!context->isCopTest() && !security_config->checkGrpcContext(grpc_context))
```
Comment (LittleFall) [inline dbms/src/Flash/FlashService.cpp:109]:
```suggestion
    if (unlikely(!context->isCopTest() && !security_config->checkGrpcContext(grpc_context)))
```

**Review by SeaRise — APPROVED** [review]

File: dbms/src/Flash/FlashService.cpp [hunk 5]
```diff
@@ -430,8 +433,8 @@ std::tuple<ContextPtr, grpc::Status> FlashService::createDBContext(const grpc::S
         std::string client_ip = peer.substr(pos + 1);
         Poco::Net::SocketAddress client_address(client_ip);
 
-        // For MPP test, we don't care about security config.
-        if (!context->isMPPTest())
+        // For MPP or Cop test, we don't care about security config.
+        if (likely(!(context->isMPPTest() || context->isCopTest())))
```
Comment (LittleFall) [inline dbms/src/Flash/FlashService.cpp:437]: can it replace to `!context->isTest()`?

**Review by LittleFall — DISMISSED** [review]

**Review by LittleFall — APPROVED** [review]
see my previous  comment, rest LGTM

File: dbms/src/Flash/FlashService.cpp (hunk 5 above)
Comment (ywqzzy) [reply dbms/src/Flash/FlashService.cpp:437]: > can it replace to `!context->isTest()`?

Yes!

**Comment by JaySon-Huang** [comment]
@LittleFall you can use "Request change" to dismiss the LGTM counting

**Pull request merged**